tests/pfcwd/test_pfcwd_timer_accuracy.py

```diff
@@ -0,0 +1,208 @@
+
+    Args:
+        setup_pfc_test (fixture): module scoped, autouse PFC fixture
+        fanout_graph_facts (fixture): fanout graph info
+        duthost (AnsibleHost): DUT instance
+        fanouthosts (AnsibleHost): fanout instance
+
+    Yields:
+        timers (dict): pfcwd timer values
+        storm_handle (PFCStorm): class PFCStorm instance
+    """
+    logger.info("--- Pfcwd timer test setup ---")
+    setup_info = setup_pfc_test
+    test_ports = setup_info['test_ports']
+    timers = setup_info['pfc_timers']
+    eth0_ip = setup_info['eth0_ip']
+    pfc_wd_test_port = test_ports.keys()[0]
+    neighbors = setup_info['neighbors']
+    fanout_info = fanout_graph_facts
+    dut = duthost
+    fanout = fanouthosts
+    peer_params = populate_peer_info(neighbors, fanout_info, pfc_wd_test_port)
+    storm_handle = set_storm_params(dut, fanout_info, fanout, peer_params)
+    timers['pfc_wd_restore_time'] = 400
+    start_wd_on_ports(dut, pfc_wd_test_port, timers['pfc_wd_restore_time'],
+                      timers['pfc_wd_detect_time'])
+    # enable routing from mgmt interface to localhost
+    dut.shell("sysctl -w net.ipv4.conf.eth0.route_localnet=1")
+    # rule to forward syslog packets from mgmt interface to localhost
+    cmd = ("iptables -t nat -I PREROUTING -p udp -d {} --dport 514 -j DNAT "
+           "--to-destination 127.0.0.1:514".format(eth0_ip))
+    dut.shell(cmd)
+
+    logger.info("--- Pfcwd Timer Testrun ---")
+    yield { 'timers' : timers,
+            'storm_handle': storm_handle
+          }
+
+    logger.info("--- Pfcwd timer test cleanup ---")
+    dut.shell("iptables -t nat --flush")
+    dut.shell("sysctl -w net.ipv4.conf.eth0.route_localnet=0")
+    storm_handle.stop_storm()
+
+def populate_peer_info(neighbors, fanout_info, port):
+    """
+    Build the peer_info map which will be used by the storm generation class
+
+    Args:
+        neighbors (dict): fanout info for each DUT port
+        fanout_info (dict): fanout graph info
+        port (string): test port
+
+    Returns:
+        peer_info (dict): all PFC params needed for fanout for storm generation
+    """
+    peer_dev = neighbors[port]['peerdevice']
+    peer_port = neighbors[port]['peerport']
+    peer_info = {'peerdevice': peer_dev,
+                 'hwsku': fanout_info[peer_dev]['device_info']['HwSku'],
+                 'pfc_fanout_interface': peer_port
+                }
+    return peer_info
+
+def set_storm_params(dut, fanout_info, fanout, peer_params):
+    """
+    Setup storm parameters
+
+    Args:
+        dut (AnsibleHost): DUT instance
+        fanout_info (fixture): fanout graph info
+        fanout (AnsibleHost): fanout instance
+        peer_params (dict): all PFC params needed for fanout for storm generation
+
+    Returns:
+        storm_handle (PFCStorm): class PFCStorm intance
+    """
+    logger.info("Setting up storm params")
+    pfc_queue_index = 4
+    pfc_frames_count = 300000
```

Comment:
What about have these as argument?

Reply:
can make pfc_queue_index an argument for this particular test but not for other tests. Is there a way to do it at the file level instead of dir level?
pfc_frames_count is not a good option because this number was arrived probably after a lot of trial and error. We need enough packets to trigger PFC storm but just enough so that the storm is not ongoing. Making this an arg would also mean we wouldn't know exactly how much time to wait for the restore to happen

Reply:
Nope, the argument are kind of global even when they are spread across multiple dirs. Attempt to ad an argument with the same name in two different dirs will result in an error.
It is up to you if you want to add an argument with new unique name for this test cases.

Reply:
will skip this for the time being